src/test/ui/print_type_sizes/niche-filling.rs

```diff
@@ -15,12 +15,19 @@
 // padding and overall computed sizes can be quite different.
 
 #![feature(start)]
+#![feature(rustc_attrs)]
 #![allow(dead_code)]
 
 use std::num::NonZeroU32;
 
 pub enum MyOption<T> { None, Some(T) }
 
+#[rustc_layout_scalar_valid_range_start(0)]
+#[rustc_layout_scalar_valid_range_end(4294967294)]
+pub struct MyNonNegativeOne {
```

Comment:
This is a nit, especially for a test type, but I keep reading this as "non-negative one" rather than "non-(negative one)", because "non" tends to combine with another word, and "non-negative" is common. Perhaps `NotNegativeOne` would be easier to parse?